### PR TITLE
[KYUUBI #708] [BUILD] Upgrade scala-maven-plugin to 4.5.3

### PR DESCRIPTION
**Pull request by wuchunfu**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@
         <maven.plugin.build.helper.version>3.2.0</maven.plugin.build.helper.version>
         <maven.plugin.compiler.version>3.8.1</maven.plugin.compiler.version>
         <!-- DO NOT bump 4.4.0, see https://github.com/NetEase/kyuubi/pull/441 -->
```
Comment (pan3793) [inline pom.xml:98]: Have a verification that #441 still exists in 4.5.3, I think we need to figure out what happens in scala maven plugin before doing this upgrading.

Comment (wuchunfu) [reply pom.xml:98]: Now I will upgrade the version of `scala-maven-plugin` to `4.5.3` and pass the local verification, and run the `build/dist --tgz` command to package normally

Comment (pan3793) [reply pom.xml:98]: Yes, it also passed ci but not the case reported by #441

Comment (wuchunfu) [reply pom.xml:98]: What should I do?

Comment (pan3793) [reply pom.xml:98]: We need to figure out what happens in scala maven plugin cause #441 first. 
If it's an accident change, fix it in scala maven plugin;
if scala maven plugin team confirme the change is by design and they don't think #441 should be supported, we can do upgrade and ignore #441.

Comment (yaooqinn) [reply pom.xml:98]: Let's add a new comment here and skip this version too.

```
<!-- DO NOT bump 4.4.0, see https://github.com/NetEase/kyuubi/pull/441 -->
<!-- DO NOT bump 4.5.3, see https://github.com/NetEase/kyuubi/issues/708 -->
```

File: pom.xml [hunk 2]
```diff
@@ -96,7 +96,8 @@
...
-        <maven.plugin.scala.version>4.3.0</maven.plugin.scala.version>
+        <!-- DO NOT bump 4.5.3, see https://github.com/NetEase/kyuubi/issues/708 -->
+        <maven.plugin.scala.version>4.5.3</maven.plugin.scala.version>
```
Comment (pan3793) [inline pom.xml:100]: revert it to `4.3.0`

**Review by turboFei — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (yaooqinn) [reply pom.xml:100]: @wuchunfu

**Review by pan3793 — APPROVED** [review]
LGTM

File: pom.xml (hunk 2 above)
Comment (wuchunfu) [reply pom.xml]: @yaooqinn I has now been revert and add a new comment here and skip this version too.

**Comment by pan3793** [comment]
Thanks, merged for v1.3.0.

**Pull request closed without merging**